wagtail_draftail_anchors/rich_text.py

```diff
@@ -1,11 +1,56 @@
+import importlib
+
+from django.conf import settings
+from django.utils.html import escape
 from draftjs_exporter.dom import DOM
 
+from wagtail import VERSION as wagtail_version
 from wagtail.admin.rich_text.converters.contentstate_models import Block
 from wagtail.admin.rich_text.converters.html_to_contentstate import (
     BlockElementHandler,
     InlineEntityElementHandler,
 )
 
+if wagtail_version >= (3, 0):
+    from wagtail.rich_text import LinkHandler
+else:
+    from wagtail.core.rich_text import LinkHandler
+
+
+# We can't use "anchor", as Wagtail uses this internally for links whose hrefs
+# start with "#"
+ANCHOR_TARGET_IDENTIFIER = "anchor-target"
+
+
+def render_span(attrs):
+    return '<span id="%s">' % escape(attrs["id"])
+
+
+def render_a(attrs):
+    id_ = escape(attrs["id"])
+    return '<a href="#%s" id="%s" data-id="%s">' % (id_, id_, id_)
```

Comment:
Nitpick: rather than escaping by hand, why not use https://docs.djangoproject.com/en/4.1/ref/utils/#django.utils.html.format_html here, which combines escaping variables passed in and marking the result as safe?